honeybee_3dm/togeometry.py

```diff
@@ -191,151 +185,155 @@ def brep_to_mesh_to_face3d(brep):
     return faces
 
 
-def brep_to_face3d(brep, tolerance):
-    """Get a Ladybug Face3D object from a planar rhino3dm Brep.
-
-    This function is used in translating rhino3dm planar Brep to Ladybug Face3D objects.
+def brep_to_face3d(brep, tolerance, obj):
+    """Get a list of Ladybug Face3D objects from a planar rhino3dm Brep.
 
     Args:
         brep: A rhino3dm Brep.
         tolerance: A rhino3dm tolerance object. Tolerance set in the rhino file.
-
+        obj: A rhino3dm object.
     Returns:
         A list of Ladybug Face3D objects.
     """
-    # Getting all vertices of the face
     mesh = brep.Faces[0].GetMesh(rhino3dm.MeshType.Any)
-    if not mesh:
-        raise AttributeError
-    # * Check 01 - If any of the edge is an arc
-    lines = []
+
+    # If any of the edge is an arc
     curved = False
     for i in range(len(brep.Edges)):
         if not brep.Edges[i].IsLinear(tolerance):
             curved = True
             break
```

Comment:
I comment on similar cases before. This is a typical use case for using **for**/else instead of creating a Boolean.

Reply:
```
animals = ['cat', 'dog', 'human', 'horse']
for animal in animals:
    print(animal)
    if animal == 'human':
        break
else:
    print(animals)
```
In the code above, `animals` will not be printed. which is what we want. My understanding is that the else clause will only be executed if the loop is finished normally. Whereas here, we want the loop to end abruptly when we find that one of the edge is curved.

Reply:
I see your point now. You are right that else won't work. In that case you really want to `return` when it happens and separate the rest of your logic after that.

```python

    for i in range(len(brep.Edges)):
        if not brep.Edges[i].IsLinear(tolerance):
            return [brep_to_meshed_face3d(brep, tolerance)]

   # the rest of the code here if the case didn't happen.

```

Reply:
Complied